flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -843,7 +843,7 @@ private float convertToFloat(Object o, float defaultValue) {
 		}
 		else if (o.getClass() == Double.class) {
 			double value = ((Double) o);
-			if (value <= Float.MAX_VALUE && value >= Float.MIN_VALUE) {
+			if (value <= Float.MAX_VALUE && value >= Float.NEGATIVE_INFINITY) {
```

Comment:
The original looks to be the correct test for underflow but should also test the negative range.